src/util.ts

```diff
@@ -79,74 +79,75 @@ export function getBuyURL(networkCode = '1', address?: string, amount = 5) {
- * @returns - URL to fetch the transactions from
- */
-export function getAlethioApiUrl(networkType: string, address: string, opt?: FetchAllOptions) {
-  if (networkType !== 'mainnet') {
-    return { url: '', headers: {} };
-  }
-  let url = `https://api.aleth.io/v1/token-transfers?filter[to]=${address}`;
-  // From alethio implementation
-  // cursor = hardcoded prefix `0x00` + fromBlock in hex format + max possible tx index `ffff`
-  let fromBlock = opt && opt.fromBlock;
-  if (fromBlock) {
-    fromBlock = parseInt(fromBlock).toString(16);
-    let prev = `0x00${fromBlock}ffff`;
-    while (prev.length < 34) {
-      prev += '0';
-    }
-    url += `&page[prev]=${prev}`;
+  if (etherscanApiKey) {
+    url += `&apikey=${etherscanApiKey}`;
   }
-  /* istanbul ignore next */
-  const headers = opt && opt.alethioApiKey ? { Authorization: `Bearer ${opt.alethioApiKey}` } : undefined;
-  return { url, headers };
+  return url;
 }
 
 /**
  * Handles the fetch of incoming transactions
  *
  * @param networkType - Network type of desired network
  * @param address - Address to get the transactions from
- * @param opt? - Object that can contain fromBlock and Alethio service API key
+ * @param opt? - Object that can contain fromBlock and Etherscan service API key
  * @returns - Responses for both ETH and ERC20 token transactions
  */
 export async function handleTransactionFetch(
   networkType: string,
   address: string,
   opt?: FetchAllOptions,
-): Promise<[{ [result: string]: [] }, { [data: string]: [] }]> {
-  const url = getEtherscanApiUrl(networkType, address, opt && opt.fromBlock);
-  const etherscanResponsePromise = handleFetch(url);
-  const alethioUrl = getAlethioApiUrl(networkType, address, opt);
-  const alethioResponsePromise = alethioUrl.url !== '' && handleFetch(alethioUrl.url, { headers: alethioUrl.headers });
-  let [etherscanResponse, alethioResponse] = await Promise.all([etherscanResponsePromise, alethioResponsePromise]);
-  if (etherscanResponse.status === '0' || etherscanResponse.result.length <= 0) {
-    etherscanResponse = { result: [] };
+): Promise<[{ [result: string]: [] }, { [result: string]: [] }]> {
+  // transactions
+  const etherscanTxUrl = getEtherscanApiUrl(
+    networkType,
+    address,
+    'txlist',
+    opt && opt.fromBlock,
+    opt && opt.etherscanApiKey,
+  );
+  const etherscanTxResponsePromise = handleFetch(etherscanTxUrl);
+
+  // tokens
+  const etherscanTokenUrl = getEtherscanApiUrl(
+    networkType,
+    address,
+    'tokentx',
+    opt && opt.fromBlock,
+    opt && opt.etherscanApiKey,
+  );
+  const etherscanTokenResponsePromise = handleFetch(etherscanTokenUrl);
+
+  let [etherscanTxResponse, etherscanTokenResponse] = await Promise.all([
+    etherscanTxResponsePromise,
+    etherscanTokenResponsePromise,
+  ]);
+  /* istanbul ignore next */
+  if (etherscanTxResponse?.status === '0' || etherscanTxResponse?.result.length <= 0) {
```

Comment:
Similar case here as well - `etherscanTxResponse` _can't_ be `undefined`, so the `?` operator shouldn't be used here.